### PR TITLE
[ADF-1733] Restore files and folders from Trash

### PR DESCRIPTION
**Pull request by pionnegru**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: docs/node-restore.directive.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+<!-- markdown-toc end -->
+
+## Basic Usage
+
+```html
+<adf-toolbar title="toolbar example">
+    <button md-icon-button
+        [restoreLocation]="/files"
+        [adf-node-restore]="documentList.selection"
```
Comment (eromano) [inline docs/node-restore.directive.md]: can we avoid the use of the kebabcase if is not a tag?

File: ng2-components/ng2-alfresco-core/src/directives/node-restore.directive.ts [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+@Directive({
+    selector: '[adf-node-restore]'
+})
+export class NodeRestoreDirective {
+    private restoreProcessStatus;
+
+    @Input('adf-node-restore')
+    selection: DeletedNodeEntry[];
+
+    @Input('restoreLocation') location: string = '';
```
Comment (eromano) [inline ng2-components/ng2-alfresco-core/src/directives/node-restore.directive.ts]: https://angular.io/guide/styleguide#avoid-aliasing-inputs-and-outputs

can we remove the aliases?

**Comment by pionnegru** [comment]
Thanks @eromano, I'll try to be more careful about minor things like this

File: ng2-components/ng2-alfresco-core/src/directives/node-restore.directive.ts [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Directive, EventEmitter, HostListener, Input, Output } from '@angular/core';
+import { MdSnackBar } from '@angular/material';
```
Comment (DenisVuyka) [inline ng2-components/ng2-alfresco-core/src/directives/node-restore.directive.ts]: Needs to be NotificaitonService from the Core library.

**Pull request merged**